index.js

```diff
@@ -76,14 +77,24 @@ exports.init = function (ssb, config) {
     }
 
     pull(
+      // we don't care about groups we've been excluded from
       ssb.tribes2.list({ live: true }),
+      pull.unique('id'),
       pull.map((group) =>
         pull(
           ssb.tribes2.listMembers(group.id, { live: true }),
-          pull.map((groupMemberId) => ({
-            groupSecret: group.secret,
-            groupMemberId,
-          }))
+          pull.map((members) => members.added),
+          pull.flatten(),
+          pull.unique(),
+          pull.map((groupMemberId) =>
+            group.readKeys
+              .map((readKey) => readKey.key)
+              .map((groupSecret) => ({
+                groupSecret,
+                groupMemberId,
+              }))
+          ),
+          pull.flatten()
```

Comment:
Is this replication setup a close enough approximation of what's needed to be in a group? Basically: all epochs from the current members.

We don't keep replicating epochs from old members, which might mean we miss some messages from those, but we also don't allow excluded members to keep trolling on their old epochs

Reply:
made a new arg for listMembers that returns every added member for every epoch